### ISSUE TITLE
Enable the command palette to be evoked everywhere in admin dashboard for WordPress 6.9

### ISSUE THREAD

**Issue by annezazu** [WordPress/gutenberg#58218]
I've heard this as a blocker for folks in adopting the command palette and, while there's a PR to address this https://github.com/WordPress/gutenberg/pull/54515, I wanted to create a corresponding issue in the repo to make it easier to gather feedback and requests. 

If this is something you'd like to see, please comment here!

**Comment by bph** [comment]
It would be great to have the Command Palette available outside the site editor as many commands could be reached faster if one doesn't have click through Appearance > Editor first. 
IE: To reach the Custom CSS, you need to be in the editor screen before invoking the Command Palette. If you just need to fix something on the custom CSS screen. 

https://github.com/WordPress/gutenberg/assets/39980/8aa498ea-44ee-425a-89f8-da55a8036a81

**Comment by hagege** [comment]
that would be great indeed :tada:

**Comment by bryanwmiller** [comment]
This would be great! When you use the command palette to access something outside of the editor (ie, Woo settings), you have to go back to the editor to use it again (which just feels confusing).

**Comment by t-hamano** [comment]
This issue was featured as "Command Palette everywhere" in [the 6.9 roadmap post](https://make.wordpress.org/core/2025/07/28/roadmap-to-6-9/#command-palette-everywhere).

I believe there are some points that need to be discussed or resolved in order to move forward with this enhancement. As a first step, I have submitted #71030, which attempts a minimal implementation.

**Comment by t-hamano** [comment]
The first implementation has shipped: #71030.

This feature will be included in the next Gutenberg 21.5, and any feedback would be appreciated. I'm particularly interested in what other new commands are needed.

**Comment by annezazu** [comment]
Amazing work @t-hamano! I'm curious too for feedback around how clear each command is, especially when there are more of them and plugins extend whats available. This touches on https://github.com/WordPress/gutenberg/issues/51505 and https://github.com/WordPress/gutenberg/issues/51742 in particular.

**Comment by t-hamano** [comment]
I'd like to discuss how far we can move this issue forward with WordPress 6.9.

However, there are many things we need to consider when exposing it to the front end, such as security concerns and building the foundation for integration with the Abilities API, which is currently underway.

In my opinion, exposing it to the frontend should be considered for a future release, and in 6.9, I would like to focus on adding new commands and improvements.

Please let us know your thoughts!

**Comment by ellatrix** [comment]
> such as security concerns

What are the security concerns for logged in users?

**Comment by t-hamano** [comment]
>What are the security concerns for logged in users?

I think it's safe if the command palette is only available to logged-in users. I was wondering if it would eventually be made available to non-logged-in users as well.

**Comment by t-hamano** [comment]
Update: To automatically register menu navigation in the dashboard, we adopted an approach that uses PHP's Menu API to automatically register commands. https://github.com/WordPress/gutenberg/pull/71476

However, when I temporarily tried using the command palette on the front end, I found that it was difficult to automatically register commands because the global `$menu` and `$submenu` were null. We'll probably end up adding a single command, something like "Go to: Dashboard" on the front end 🤔

cc @youknowriad

**Comment by youknowriad** [comment]
@t-hamano That sounds good to me, I'm not entirely convinced that we need to have the command palette in the frontend at all.

**Comment by t-hamano** [comment]
I thought "everywhere" included the front end as well 😅 In fact, in the initial experiment #54515, it seems that we were aiming to expose the command palette in the front end as well.

**Comment by ramonjd** [comment]
> exposing it to the frontend should be considered for a future release

I can imagine possibilities down the track for this, e.g., for shortcuts to admin areas, even making targeted changes to content or styles with the help of an agent.

In the short term, and for 6.9, I'd submit a narrower scope is preferable to sort out any bugs and take away a bit of pressure. Let the idea stew and something good will come of it later on I'd say. What do folks think?

**Comment by annezazu** [comment]
I agree. I think having it in the Admin along is a big step and a future release could add it to the frontend.

**Comment by t-hamano** [comment]
Thanks for the feedback, everyone!

> In the short term, and for 6.9, I'd submit a narrower scope is preferable to sort out any bugs and take away a bit of pressure. Let the idea stew and something good will come of it later on I'd say. What do folks think?

I agree. Let's mark this issue as "done" for 6.9 and move it to a future release. Improvements to the Command Palette itself can be tracked in #66648.

**Comment by ellatrix** [comment]
> I think it's safe if the command palette is only available to logged-in users. I was wondering if it would eventually be made available to non-logged-in users as well.

I don't see how it's expected for non-logged-in users. Why not add it for logged-in users if it's safe for them? If we're advertising that you can use the command palette everywhere, users will probably expect it any time they see the admin bar.

**Comment by ellatrix** [comment]
Btw, something cool to try for the logged-in front-end could be to lazy load the script after you press cmd+k :)

**Comment by t-hamano** [comment]
> something cool to try for the logged-in front-end could be to lazy load the script after you press cmd+k :)

Perhaps we enqueue the command as a script module and then dynamically import the script using Ctrl+K as a trigger? I'm not very knowledgeable in that area, so I might be wrong.

**Comment by annezazu** [comment]
@t-hamano  why is this listed as "punted to 7.0" in the 6.9 board? I'm going to re-name this for now and mark it as an iteration issue for 6.9. From there, we can close this out and mark it as "done" for 6.9 IMO, unless I'm missing something :)

**Comment by annezazu** [comment]
Closing this for now but I'll leave it to you to update the status in the 6.9 board 🙇🏼

**Comment by t-hamano** [comment]
> why is this listed as "punted to 7.0" in the 6.9 board?

@annezazu If we plan to enable the command palette on the front end as well,  I  thought it would be good to leave this issue open and address it in the future. But for now, we can consider this issue done 👍

**Comment by bph** [comment]
>why is this listed as "punted to 7.0" in the 6.9 board? I'm going to re-name this for now and mark it as an iteration issue for 6.9. From there, we can close this out and mark it as "done" for 6.9 IMO, unless I'm missing something :)

Thanks for clarifying, @annezazu—I was already getting confused, what will make or not make it into 6.9

**Comment by MatzeKitt** [comment]
What’s the best practice now to include custom commands within the whole admin? Before, you could just register the commands and load them inside the editor. Now, do you need to manually enqueue your script(s) with custom commands within `admin_enqueue_scripts`?

**Comment by t-hamano** [comment]
I think `admin_enqueue_scripts` is fine.

The core function also uses this hook: https://github.com/WordPress/wordpress-develop/blob/9f0e3b6c4f875c5825619066fd8ec6a59569ffd9/src/wp-includes/default-filters.php#L601

**Comment by proweb** [comment]
So if I no need for Command Palette everywhere how to disable it in wp-admin? (which hook and code)

**Comment by t-hamano** [comment]
@proweb Try this code.

```php
add_action( 'init', function() {
	remove_action( 'admin_enqueue_scripts', 'wp_enqueue_command_palette_assets' );
	remove_action( 'admin_enqueue_scripts', 'gutenberg_enqueue_command_palette_assets', 9 );
});
```

**Comment by proweb** [comment]
> [@proweb](https://github.com/proweb) Try this code.
> 
> add_action( 'init', function() {
> 	remove_action( 'admin_enqueue_scripts', 'wp_enqueue_command_palette_assets' );
> 	remove_action( 'admin_enqueue_scripts', 'gutenberg_enqueue_command_palette_assets', 9 );
> });

Thanks it's work in 7.0
Now i need just remove `Ctrl+K` in editor-document-bar but I think it's not big problem.

**Comment by t-hamano** [comment]
> Now i need just remove `Ctrl+K` in editor-document-bar

Try this code.

```php
add_action( 'admin_bar_menu', function () {
	remove_action( 'admin_bar_menu', 'wp_admin_bar_command_palette_menu', 55 );
	remove_action( 'admin_bar_menu', 'gutenberg_admin_bar_command_palette_menu', 55 );
}, 50 );
```
